bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrubyscripting.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.apache.felix.service.command.Process;
+import org.apache.karaf.shell.api.console.Command;
+import org.apache.karaf.shell.api.console.CommandLine;
+import org.apache.karaf.shell.api.console.Completer;
+import org.apache.karaf.shell.api.console.Parser;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.karaf.shell.api.console.SessionFactory;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.automation.jrubyscripting.internal.watch.JRubyScriptFileWatcher;
+import org.openhab.core.automation.module.script.ScriptEngineManager;
+import org.openhab.core.io.console.Console;
+import org.openhab.core.io.console.StringsCompleter;
+import org.openhab.core.scheduler.ScheduledCompletableFuture;
+import org.openhab.core.scheduler.Scheduler;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JRubyConsoleCommandExtension} class
+ *
+ * @author Jimmy Tanagra - Initial contribution
+ */
+@Component(immediate = true)
+// @NonNullByDefault cannot be applied to classes that implement Command
+// This is implemented directly using the Command interface in order to access Karaf's Session
+public class JRubyConsoleCommandExtension implements Command, Completer {
+    private final Logger logger = LoggerFactory.getLogger(JRubyConsoleCommandExtension.class);
+
+    private static final String INFO = "info";
+    private static final String SHELL = "shell";
```

Comment:
is this really what we want to call it? openHAB's concept is called console (`openhab-cli console`), Bundler's console is called console (`bundle console`), the preferred binstub in Ruby is `bin/console`. There's a weaker argument for calling it `irb`, but given that it's not currently backed by IRB, and the other precedents, I strongly lean towards calling it `console` instead of `shell`.

Reply:
console is a great name. I'll rename to it.